src/overlay/OverlayManagerImpl.cpp

```diff
@@ -282,16 +272,17 @@ OverlayManagerImpl::tick()
     filterByNextAttempt(peers);
     connectTo(peers);
 
-    if (getAuthenticatedPeersCount() < mApp.getConfig().TARGET_PEER_CONNECTIONS)
+    auto outboundCount =
+        mOutboundPendingPeersCount + mOutboundAuthenticatedPeersCount;
+    if (outboundCount < mApp.getConfig().TARGET_PEER_CONNECTIONS)
```

Comment:
Yeah, I think it's better to keep things separate like before:
* `MAX_*_PENDING_CONNECTIONS` should only limit pending connections.
* `TARGET_*` limits actual connections
so total number of connections is the *sum*, interestingly the way we clamp MAX_PENDING_CONNECTIONS in config.cpp seems wrong because of that, I think the code should be (in current master terminology):
```c++
        // ensure that max pending connections is not above what the system
        // supports
        int max_pending = std::min<int>(MAX_PENDING_CONNECTIONS,
                fs::getMaxConnections()-MAX_PEER_CONNECTIONS);
        if (max_pending <= 0 || max_pending > UINT16_MAX)
        {
            throw std::invalid_argument(
                "invalid MAX_PEER_CONNECTIONS/MAX_PENDING_CONNECTIONS");
        }

        MAX_PENDING_CONNECTIONS = static_cast<unsigned short>(max_pending);

        // enforce TARGET_PEER_CONNECTIONS <= MAX_PEER_CONNECTIONS <=
        // MAX_PENDING_CONNECTIONS
        MAX_PEER_CONNECTIONS =
            std::min(MAX_PEER_CONNECTIONS, MAX_PENDING_CONNECTIONS);
        TARGET_PEER_CONNECTIONS =
            std::min(TARGET_PEER_CONNECTIONS, MAX_PEER_CONNECTIONS);
```


I thought about which values make sense for this: we can simply set
`MAX_OUTBOUND_PENDING_CONNECTIONS = 2*TARGET_PEER_CONNECTIONS`
and
`MAX_INBOUND_PENDING_CONNECTIONS = MAX_PENDING_CONNECTIONS-MAX_OUTBOUND_PENDING_CONNECTIONS`
(of course ensuring that the resulting `MAX_INBOUND_PENDING_CONNECTIONS > 0`)

Reply:
Reason I prefer to completely separate inbound and outbound parameters and tracking is that even if a node is getting DoSed by inbound connections, it can still perform outbound connections to other nodes